app/master/build_scheduler.py

```diff
@@ -112,16 +114,17 @@ def execute_next_subjob_or_free_executor(self, slave):
                 subjob = self._build._unstarted_subjobs.get(block=False)
                 self._logger.debug('Sending {} to {}.', subjob, slave)
                 try:
-                    slave.start_subjob(subjob)
-                    subjob.mark_in_progress(slave)
+                    if self._build.is_canceled:
+                        self._free_slave_executor(slave)
+                    else:
+                        slave.start_subjob(subjob)
+                        subjob.mark_in_progress(slave)
 
                 except SlaveError as ex:
                     internal_errors.labels(ErrorType.SubjobWriteFailure).inc()  # pylint: disable=no-member
                     self._logger.warning('Failed to start {} on {}: {}. Requeuing subjob and freeing slave executor...',
                                          subjob, slave, repr(ex))
-                    self._build._unstarted_subjobs.put(subjob)  # todo: This changes subjob execution order. (Issue #226)
-                    # An executor is currently allocated for this subjob in begin_subjob_executions_on_slave.
-                    # Since the slave has been marked for shutdown, we need to free the executor.
```

Comment:
I'm fine with removing the todo since there is an issue filed for it but should we keep the comment (in some form -- we could remove the reference to begin_subjob_executions_on_slave)?